### PR TITLE
updated mdf.py to enable it to generate condition and condition set documentation

### PR DESCRIPTION
**Pull request by Monsurat-Onabajo**

### CODE REVIEW

**Comment by Monsurat-Onabajo** [comment]
the tests are failing because the file path is not recognized, but it is working perfectly fine on my local machine, i dont know what could be wrong

File: src/modeci_mdf/mdf.py [hunk 1]
```diff
@@ -68,7 +68,7 @@ class Function(MdfBase):
             https://mdf.readthedocs.io/en/latest/api/MDF_function_specifications.html
         args: Dictionary of values for each of the arguments for the Function, e.g. if the in-built function
               is linear(slope),the args here could be {"slope":3} or {"slope":"input_port_0 + 2"}
-        value: If the function is a value expression, this attribute will contain the expression and the function
+        valuess: If the function is a value expression, this attribute will contain the expression and the function
```
Comment (sanjayankur31) [inline src/modeci_mdf/mdf.py:71]: just checking: is it `valuess` (two `s`?)

File: src/modeci_mdf/mdf.py [hunk 2]
```diff
@@ -613,3 +613,24 @@ def new_unstructure(obj):
         pass
     else:
         converter.register_unstructure_hook(v, parsed_unstructure_factory(v))
+
+model = Model(id="test")
```
Comment (sanjayankur31) [inline src/modeci_mdf/mdf.py:617]: Should all of this be here, or in `generate.py` perhaps?

**Comment by sanjayankur31** [comment]
As noted on Slack, the tests are failing because `pytest` runs from the main `MDF` directory, and the path in `mdf.py` looks like it is relative to the folder that contains `mdf.py`. But, the main issue is that the code to generate the docs should not go into `mdf.py`---if it's there, it'll get run each time the `mdf.py` module is imported. So, I think the new code you've added should go into `docs/generate.py`. This file is not to be imported by users, and so won't be used in the test.

File: src/modeci_mdf/mdf.py (hunk 1 above)
Comment (Monsurat-Onabajo) [reply src/modeci_mdf/mdf.py:71]: i will edit it now

**Comment by pgleeson** [comment]
Thanks @Monsurat-Onabajo! Looks good.

**Pull request merged**